### ISSUE TITLE
Update to React 18

### ISSUE THREAD

**Issue by oliwerhelsen** [supabase-community/supabase-auth-helpers#64]
Hi Guys,

Lovely helper! Could we add support for React 18? Or can I contribute and push the changes myself. 

Best regards
Oliwer H.

**Comment by alaister** [comment]
Hi Oliwer,

PRs are always very welcome!

What doesn't work with React 18 at the moment? Or are you just looking for Suspense support?

Thanks!

**Comment by oliwerhelsen** [comment]
Getting a mismatch problem with the new version of NextJS.

**Comment by mirkoisaia** [comment]
To be precise    
` "@supabase/ui": "^0.36.3",`
is the culprit, as it requires 
`react@"^16.13.1 || ^17.0.1" from @supabase/ui@0.36.4`

if you clone the repository of auth-helpers and then proceed to remove supabase/ui from dependencies, then build and it works. 

it will be missing supabase ui, but a force npm install closed the circle for me.

**Comment by thorwebdev** [comment]
This relates to #22 we need to get a better separation set up.

**Comment by Murkrage** [comment]
> To be precise ` "@supabase/ui": "^0.36.3",` is the culprit, as it requires `react@"^16.13.1 || ^17.0.1" from @supabase/ui@0.36.4`
> 
> if you clone the repository of auth-helpers and then proceed to remove supabase/ui from dependencies, then build and it works.
> 
> it will be missing supabase ui, but a force npm install closed the circle for me.

I'm wondering why `@supabase/ui` is required for the auth helpers. Can't seem to find a reference of it in the auth helpers at all.

**Comment by mirkoisaia** [comment]
as far as i can tell it should be involved with the aesthetic of the login page, as when i tried it, even without setting any other code or css, it was already well designed and placed in my screen with the google icon

**Comment by Murkrage** [comment]
I'm not sure if that makes sense for an auth helper, to be honest. I'd expect it to be UI agnostic. Since `@supabase/ui` exists I would expect an easy API on that side to get it all to work.

**Comment by Murkrage** [comment]
@mirkoisaia I've tried removing `@supabase/ui` from a local clone of the repo but I'm getting a bunch of type errors. What did you do to get it to work locally? (Whilst we wait for this to be fixed)

**Comment by mirkoisaia** [comment]
if i'm not wrong i just removed the link to supabase/ui in package.json, built the cloned repo but without supabase/ui, installed that (overwriting the one in nodemodule of my project) , without react warning as i was installing, then npm i @supabase/ui or what the package actual name was. 

also not sure if it may be the case i was on plain javacript not ts

**Comment by sakulstra** [comment]
As this is currently breaking for anyone using react 18, wouldn't it be reasonable to widen the version range on https://github.com/supabase/ui/blob/master/package.json#L21 to unbreak the apps?

**Comment by thorwebdev** [comment]
The UI part will be spun out into its own package soon: https://github.com/supabase-community/supabase-auth-helpers/issues/22#issuecomment-1096445631

**Comment by daanvosdewael** [comment]
The `react` and `react-dom` versions specified in the [`nextjs`](https://github.com/supabase-community/supabase-auth-helpers/blob/b6cbee5f08059df67c7f66e76b969d45cfcda036/packages/nextjs/package.json) and [`react`](https://github.com/supabase-community/supabase-auth-helpers/blob/b6cbee5f08059df67c7f66e76b969d45cfcda036/packages/react/package.json) packages differ. One is v18, the other still v17. Currently this breaks with a Next.js project on v18. Getting the following error:
```
Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
```

This is without `@supabase/ui`.

**Comment by mirkoisaia** [comment]
> The `react` and `react-dom` versions specified in the [`nextjs`](https://github.com/supabase-community/supabase-auth-helpers/blob/b6cbee5f08059df67c7f66e76b969d45cfcda036/packages/nextjs/package.json) and [`react`](https://github.com/supabase-community/supabase-auth-helpers/blob/b6cbee5f08059df67c7f66e76b969d45cfcda036/packages/react/package.json) packages differ. One is v18, the other still v17. Currently this breaks with a Next.js project on v18. Getting the following error:
> 
> ```
> Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
> 1. You might have mismatching versions of React and the renderer (such as React DOM)
> 2. You might be breaking the Rules of Hooks
> 3. You might have more than one copy of React in the same app
> See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
> ```
> 
> This is without `@supabase/ui`.

Had to admin that **I DON'T KNOW WHAT I'M DOING**, as I'm a self taught in the process of learning all this new sparlking toys. but i managed to get behind that error this way: 
- downloading the release (not installing it with npm, and successive installation via npm will reset this process, so you can imagine how useful it is not.) 
- installing and building the release
- erasing the content of not working already installed auth helpers, and replacing it with the dist of the freshly built 1.3.0. 
- that way it works, but it open to this [https://github.com/supabase-community/supabase-auth-helpers/issues/84](url)
- that you can solve with this: [https://github.com/supabase-community/supabase-auth-helpers/issues/84#issuecomment-1111430713](url)
- i'm not sure cause i was switching between yarn add to avoid npm install process, but the last (resolution) may have fixed the first error. I didn't tried, to be honest, as I said i was just playing with it.

hope it's useful. :)

**Comment by Murkrage** [comment]
@thorwebdev Any update on a possible release with the changes to the monorepo? It's been a month since the issue of a version mismatch with React was reported (this ticket) and I'm really looking forward to getting the auth helper in my project since it's the recommended way to handle auth moving forward 😄

**Comment by silentworks** [comment]
We are aiming to get a release out this week but we just need to document some changes as the library name is changing along with the versioning.

**Comment by mansoorsiddiqui** [comment]
Any more updates on the React 18 release in terms of specific timeline?

**Comment by johnbellone** [comment]
Anyone got a workaround here?

**Comment by mansoorsiddiqui** [comment]
> Anyone got a workaround here?

Currently, I got auth set up and working with React 17, then commented out the auth so I could upgrade to React 18 and work on the rest of my project, hoping that they will release the update by the time I finish the rest. I tried a million other things like building the release myself with parts changed / removed etc. and this was the best strategy I could come up with.
